src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs

```diff
@@ -96,5 +116,31 @@ public static ILoggingBuilder AddOpenTelemetryLogs(this ILoggingBuilder builder)
 
         return builder;
     }
+
+    private static bool IsAutoInstrumentationStartupAssemblyConfigured()
```

Comment:
Why do we need this method? We could directly call `IsAutoInstrumentationStartupAssemblySet`.
Additionally, the comment in this method might be incorrect. Bytecode instrumentation is called before HostingStartupAssembly, so the integration might not have occurred yet. Since HostingStartupAssembly is set, we expect `OpenTelemetry.AutoInstrumentation.AspNetCoreBootstrapper` to handle the logger integration at a later time.

Reply:
This is here to ensure we check environment variable once, assuming we want to support scenarios as described in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3003#discussion_r1358171530

Reply:
Checking an environment variable is not a big cost, better to simplify the code here.

Reply:
removed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3003/commits/4d7b009fc77838ae84c72345bd409bd20dbca44f